### ISSUE TITLE
x509: certificate signed by unknown authority when using self-signed certificate

### ISSUE THREAD

**Issue by KaptinLin** [ainsleyclark/go-mail#20]
**Is your feature request related to a problem? Please describe.**
I am using [mailpit](https://github.com/axllent/mailpit/wiki/SMTP-with-STARTTLS-and-authentication#smtp-authentication) for mail testing, but i got `x509: certificate signed by unknown authority` error

**Describe the solution you'd like**
Maybe add `InsecureSkipVerify: true` config when testing

**Comment by ainsleyclark** [comment]
Have you tried passing a different http client to the default one in the constructor? Not sure adding this functionality to the client is secure. Let me know how you get on

**Comment by KaptinLin** [comment]
Yes, i can pass a different [http client](https://github.com/ainsleyclark/go-mail/blob/6ef3bf60a384954861fac6720a6a7de1f0fad374/mail/config.go#L32). But it seems smtp using [smtp.SendMail](https://github.com/ainsleyclark/go-mail/blob/6ef3bf60a384954861fac6720a6a7de1f0fad374/drivers/smtp.go#L58) instead of Client send request.
You may see https://mimepost.com/blog/sending-an-email-with-golang/

Maybe you can add some code to [smtp driver](https://github.com/ainsleyclark/go-mail/blob/647242951279d3a79e9db670d72d5e78a5e63826/drivers/smtp.go#L72) to check if Client exists, then using client like [this](https://gist.github.com/chrisgillis/10888032#file-ssl_smtp_example-go-L55).

**Comment by ainsleyclark** [comment]
That will be easy to do. I'll add it in soon. Thanks for the feedback!
